librz/include/rz_analysis.h

```diff
@@ -1435,7 +1435,54 @@ RZ_API void rz_analysis_esil_interrupts_fini(RzAnalysisEsil *esil);
 RZ_API void rz_analysis_esil_mem_ro(RzAnalysisEsil *esil, int mem_readonly);
 RZ_API void rz_analysis_esil_stats(RzAnalysisEsil *esil, int enable);
 
-/* trace */
+/* new trace implementation */
+typedef enum {
+	TRACE_READ, ///< read
+	TRACE_WRITE ///< write
+} TRACE_OPERATION;
+
+typedef struct {
+	ut64 addr; ///< memory address
+	TRACE_OPERATION behavior; ///< read or write, see TRACE_OPERATION enums
+	ut64 value; ///< data either written to or read from
```

Comment:
it might be wrong to assume ut64. For example, instructions that work on XMM/YMM registers in x86_64 might access 128bits in one shot or more probably. Maybe we should have `ut8 value[32]`.

Reply:
About this `value` field, I checked the original code. Though it store the buffer data into SDB, it still read from it convert to ut64 type by using `sdb_get_num`. I think we can use the buffer and keep the logic of returning a `ut64` value when get from mem_op_trace for now, what do you think ?